sdk/runtime/metrics.go

```diff
@@ -0,0 +1,77 @@
+package runtime
+
+import (
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// MustExportPrometheus starts an HTTP server with the Prometheus handler.
+// It starts on a random open port and returns the listener. It is the caller
+// responsability to close the listener.
+func (re *RunEnv) MustExportPrometheus() net.Listener {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(err)
+	}
+
+	go http.Serve(listener, promhttp.Handler())
+	return listener
+}
+
+// HTTPPeriodicSnapshots periodically fetches the snapshots from the given address
+// and outputs them to the out directory. Every file will be in the format timestamp.out.
+func (re *RunEnv) HTTPPeriodicSnapshots(ctx context.Context, addr string, dur time.Duration, outDir string) error {
+	if !strings.HasPrefix(addr, "http") {
```

Comment:
I think it would be better to have a strict contract here.

Expect a proper endpoint in this function, and just pass http:// + listener.Addr().String() to it. In the future we might pass an https etc.

Reply:
Done! I agree

Reply:
In the future we could make this function more generic by taking an `*http.Request`, which can be reused safely (apparently stores no state: https://github.com/golang/go/issues/19653). Then this func would make no assumptions about the HTTP request method, headers, body, etc. For now, we're fine like this 👍